### PR TITLE
feat: forgot password endpoint

### LINKED ISSUE #12: [FEAT]: Authentication - Forget and Reset Password
### Description
Develop a forgot password feature for our platform to enable users to reset their passwords securely in case they forget them. This feature should include a user-friendly password recovery process that guides users through resetting their passwords.
### Acceptance Criteria
- [ ]  Verified that the user actually exists on record
- [ ] Newly created password hashed
### Expected Outcome
- User should be able to initiate a 'reset password' request, and immediately receive a password reset link once the request is made
### Requirements
- Button / Link for 'Forgot Password'
- Notification for user to check their email for a reset link
- Verify user exists
- Encrypt new password
### **Forgot Password [POST] `/api/v1/auth/forgot-password`**
1. Forgot Password Endpoint
- Given a request with valid details (ie `email`), when the user clicks on Forgot Password, then the system should send a password reset link to the email provided with a 200 status code.
2. Incorrect Email
- Given a email does not exists, when the user tries to forget password, then the system should return a 400 bad request error status with an appropriate error message
**Request**
```
POST /api/v1/auth/forgot-password
{
  "email": String
}
```
**Successful Response**
```
{
  "message": String,
}
```
**Error Response**
```
{
  "message": String,
  "error": String,
  "status_code": Int,
}
```
### **Reset Password [POST] `/api/v1/auth/reset-password/{token}`**
1. Reset Password Endpoint
- Given a request with valid details (ie `password`), when the user Reset Password, then the system should respond with a 200 status code on successful password reset.
**Request**
```
POST /api/v1/auth/reset-password/{token}
{
  "password": String
}
```
**Successful Response**
```
{
  "message": String,
}
```
**Error Response**
```
{
  "message": String,
  "error": String,
  "status_code": Int,
}
```

### PR DESCRIPTION
**Pull request by alexindevs**
## Description

This pull request adds the forgot password functionality to the authentication module. It includes the following changes:

- Added `forgotPassword` method in `AuthService`.
- Added `forgotPassword` endpoint in `AuthController`.
- Added necessary DTOs for forgot password.
- Updated tests for the new functionality.

## Changes

1. Added a new method in `AuthService` to handle forgot password logic.
2. Updated `AuthController` to include the forgot password endpoint.
3. Created DTOs for forgot password requests.
4. Added unit tests for the `forgotPassword` method in `AuthService`.

## Checklist

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes.

## Related Issues

Fixes #12

### CODE REVIEW

File: .gitignore
Comment (Heba-WebDev) [inline .gitignore:1]: Why is .env.local here?

File: src/database/data-source.ts
Comment (Heba-WebDev) [inline src/database/data-source.ts:1]: Why was this changes? Please let it be as it was before

File: src/modules/auth/auth.controller.ts
Comment (Heba-WebDev) [inline src/modules/auth/auth.controller.ts:1]: Please return the nestjs way, look at the docs (or merged PRs).

**Review by Heba-WebDev — CHANGES_REQUESTED** [review]
I left few comments please fix them

File: .gitignore
Comment (alexindevs) [reply .gitignore:1]: I've removed it.

File: src/database/data-source.ts
Comment (alexindevs) [reply src/database/data-source.ts:1]: I've reverted.

File: src/modules/auth/auth.controller.ts
Comment (alexindevs) [reply src/modules/auth/auth.controller.ts:1]: I've fixed it; please review.

File: src/modules/otp/otp.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { getRepositoryToken } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { OtpService } from './otp.service';
+import { Otp } from './entities/otp.entity';
+import { User } from '../user/entities/user.entity';
+
+// Mock the generateSixDigitToken function
```
Comment (Cobidex) [inline src/modules/otp/otp.service.spec.ts:8]: Dangling comment, best removed

File: src/modules/user/user.service.ts [hunk 2]
```diff
@@ -13,11 +13,11 @@ export default class UserService {
     private userRepository: Repository<User>
   ) {}
 
-  async createUser(user: CreateNewUserOptions) {
+  async createUser(user: CreateNewUserOptions): Promise<any> {
     const newUser = new User();
     Object.assign(newUser, user);
     newUser.is_active = true;
-    await this.userRepository.save(newUser);
+    return await this.userRepository.save(newUser);
```
Comment (Cobidex) [inline src/modules/user/user.service.ts:20]: No need for await keyword when returning a promise from function. Reduces response time needlessly

**Review by Cobidex — COMMENTED** [review]
Good Job! Few comments

**Comment by Cobidex** [comment]
LGTM

**Review by Cobidex — APPROVED** [review]

**Review by Heba-WebDev — APPROVED** [review]
All good now

**Comment by Heba-WebDev** [comment]
@alexindevs Please resolve the conflicts

**Comment by alexindevs** [comment]
On it!

**Review by TheCodeGhinux — APPROVED** [review]

File: .env.local [hunk 3]
```diff
@@ -17,4 +17,30 @@
 PROFILE=local
 NODE_ENV=development
 
-PORT=3008
\ No newline at end of file
+
+PORT=3008
+SMTP_HOST=""
```
Comment (buka4rill) [inline .env.local:22]: remove quotes

File: .env.local [hunk 4]
```diff
@@ -17,4 +17,30 @@
...
+SMTP_PORT=587
+SERVER_NAME="Boilerplate"
+SMTP_USER=""
+SMTP_PASSWORD=""
```
Comment (buka4rill) [inline .env.local:26]: remove quotes

**Review by buka4rill — COMMENTED** [review]
few more comments

File: .env.local [hunk 5]
```diff
@@ -17,4 +17,30 @@
 PROFILE=local
 NODE_ENV=development
 
-PORT=3008
\ No newline at end of file
+
+PORT=3008
+SMTP_HOST=
+SMTP_PORT=587
+SERVER_NAME="Boilerplate"
```
Comment (buka4rill) [inline .env.local:24]: Remove quotes

**Review by buka4rill — COMMENTED** [review]
another one

**Review by buka4rill — APPROVED** [review]

File: src/modules/user/user.service.ts [hunk 6]
```diff
@@ -25,11 +25,11 @@ export default class UserService {
     private userRepository: Repository<User>
   ) {}
 
-  async createUser(user: CreateNewUserOptions) {
+  async createUser(user: CreateNewUserOptions): Promise<any> {
     const newUser = new User();
     Object.assign(newUser, user);
     newUser.is_active = true;
-    await this.userRepository.save(newUser);
+    return this.userRepository.save(newUser);
```
Comment (Homoakin619) [inline src/modules/user/user.service.ts:32]: remove return statement from here.  The return value is not being used

**Pull request merged**